include/cuco/detail/static_map.inl

```diff
@@ -263,24 +328,25 @@ __device__ bool static_map<Key, Value, Scope, Allocator>::device_mutable_view::i
     key_type const existing_key = current_slot->first.load(cuda::std::memory_order_relaxed);
     // The user provide `key_equal` can never be used to compare against `empty_key_sentinel` as the
     // sentinel is not a valid key value. Therefore, first check for the sentinel
-    auto const slot_is_empty =
-      detail::bitwise_compare(existing_key, this->get_empty_key_sentinel());
+    auto const slot_is_available =
+      detail::bitwise_compare(existing_key, this->get_empty_key_sentinel()) or
+      detail::bitwise_compare(existing_key, this->get_erased_key_sentinel());
 
     // the key we are trying to insert is already in the map, so we return with failure to insert
-    if (not slot_is_empty and key_equal(existing_key, insert_pair.first)) { return false; }
+    if (not slot_is_available and key_equal(existing_key, insert_pair.first)) { return false; }
 
-    if (slot_is_empty) {
+    if (slot_is_available) {
       auto const status = [&]() {
         // One single CAS operation if `value_type` is packable
         if constexpr (cuco::detail::is_packable<value_type>()) {
-          return packed_cas(current_slot, insert_pair, key_equal);
+          return packed_cas(current_slot, insert_pair, key_equal, existing_key);
         }
 
         if constexpr (not cuco::detail::is_packable<value_type>()) {
 #if __CUDA_ARCH__ < 700
-          return cas_dependent_write(current_slot, insert_pair, key_equal);
+          return cas_dependent_write(current_slot, insert_pair, key_equal, existing_key);
 #else
-          return back_to_back_cas(current_slot, insert_pair, key_equal);
+          return back_to_back_cas(current_slot, insert_pair, key_equal, existing_key);
 #endif
         }
       }();
```

Comment:
How about moving this lambda to a function named `device_mutable_view::compare_and_swap` to hide cuda arch and packability details from the public:
```cpp
insert_result compare_and_swap(
  iterator current_slot,
  value_type const& insert_pair,
  KeyEqual key_equal,
  Key expected_key) noexcept
{
  // One single CAS operation if `value_type` is packable
  if constexpr (cuco::detail::is_packable<value_type>()) {
     return packed_cas(current_slot, insert_pair, key_equal, expected_key);
  }
  if constexpr (not cuco::detail::is_packable<value_type>()) {
#if __CUDA_ARCH__ < 700
     return cas_dependent_write(current_slot, insert_pair, key_equal, expected_key);
#else
     return back_to_back_cas(current_slot, insert_pair, key_equal, expected_key);
  }
}
```
This will clean up `insert` (`erase` also probably?) implementations a lot.